util/utils.go

```diff
@@ -42,14 +42,16 @@ const fetchSizeLimit = 1024 * 1024 * 512
 func ApiEndpoint(ipfspath string) (string, error) {
 	apifile := filepath.Join(ipfspath, "api")
 
-	val, err := ioutil.ReadFile(apifile)
+	valbytes, err := ioutil.ReadFile(apifile)
 	if err != nil {
 		return "", err
 	}
 
-	parts := strings.Split(string(val), "/")
+	val := strings.TrimRight(string(valbytes), "\n\r")
```

Comment:
`\n\r` - should we just trim all whitespace, given that [newline representation](https://en.wikipedia.org/wiki/Newline#Representation) can differ by OS?

Reply:
Hold up, if I read the docs for `TrimRight` I think this doesn't matter, it doesn't trim that string, it trims all characters in that string.

Reply:
Well, `\n\r` will trim all CR/LF in any order. However, there's no reason not to just trim all whitespace.

Reply:
(I changed it to just trim all whitespace anyways as that's nicer).